cmd/process-agent/config.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func getChecks(sysCfg *sysconfig.Config, oCfg *oconfig.OrchestratorConfig, canAccessContainers bool) (checkCfg []checks.Check) {
+	rtChecksEnabled := !ddconfig.Datadog.GetBool("process_config.disable_realtime_checks")
+	if ddconfig.Datadog.GetBool("process_config.process_collection.enabled") {
+		checkCfg = append(checkCfg, checks.Process)
+	} else {
+		if ddconfig.Datadog.GetBool("process_config.container_collection.enabled") && canAccessContainers {
+			checkCfg = append(checkCfg, checks.Container)
+			if rtChecksEnabled {
+				checkCfg = append(checkCfg, checks.RTContainer)
+			}
+		}
+		if ddconfig.Datadog.GetBool("process_config.process_discovery.enabled") {
+			checkCfg = append(checkCfg, checks.ProcessDiscovery)
+		}
+	}
+
+	// activate the pod collection if enabled and we have the cluster name set
+	if oCfg.OrchestrationCollectionEnabled {
+		if oCfg.KubeClusterName != "" {
+			checkCfg = append(checkCfg, checks.Pod)
+		} else {
+			_ = log.Warnf("Failed to auto-detect a Kubernetes cluster name. Pod collection will not start. To fix this, set it manually via the cluster_name config option")
+		}
+	}
+
+	if sysCfg.Enabled {
```

Comment:
don't we need to enable the other system-probe checks?
https://github.com/DataDog/datadog-agent/blob/9925e508eec861aa809668e2dcf145bacda27174/pkg/process/config/config.go#L332-L337
https://github.com/DataDog/datadog-agent/blob/9925e508eec861aa809668e2dcf145bacda27174/cmd/system-probe/config/config.go#L38-L42

Reply:
Nope! The only modules that are actually used by the process agent are the `PrcessModule` (which enables calling out to the system-probe to collect privileged stats) and the `NewtorkTracerModule` (which enables the connections check). The rest (`TCPQueueLengthCheckName`, `OOMKillCheckName`, and `NetworkCheckName`) are never referenced in the process agent, other than when it is added to `EnabledChecks`

Reply:
Oh, that's a good point! Thanks for clarifying it!
We should probably add `team/networks` as code reviewers as well since this change affects them.